packages/typescript-estree/src/create-program/createIsolatedProgram.ts

```diff
@@ -65,6 +65,9 @@ function createIsolatedProgram(
   const program = ts.createProgram(
     [parseSettings.filePath],
     {
+      ...(ts.JSDocParsingMode && {
```

Comment:
Nit cos the previous version didn't check this prop couldn't this just be this (without the spread)

```ts
        jsDocParsingMode: ts.JSDocParsingMode?.ParseForTypeInfo,
```